### PR TITLE
Meeting room booking

### PR DESCRIPTION
**Pull request by hi-dima**

### CODE REVIEW

File: client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      this.mode = 'update';
+    } else {
+      this.defaults = {} as MeetingRoom;
+    }
+
+    this.companyProjectsService.getAll().subscribe(companyProjects => {
+      this.projectsOptions = companyProjects
+    });
+
+    this.form = this.fb.group({
```
Comment (aiandev) [inline client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts]: use validations of `FormGroup`

File: client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  createMeetingRoom() {
+    const meetingRoom = this.form.value;
+    this.dialogRef.close(meetingRoom);
+  }
+
+  updateMeetingRoom() {
+    const meetingRoom = this.form.value;
+    meetingRoom.id = this.defaults.id;
```
Comment (aiandev) [inline client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts]: ??? I have no idea what this code here

File: client-side/src/app/_pages/meeting-room/meeting-room.component.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
+import {
+  AfterViewInit,
+  Component,
+  Input,
```
Comment (aiandev) [inline client-side/src/app/_pages/meeting-room/meeting-room.component.ts]: inline import please

File: client-side/src/app/_pages/meeting-room/meeting-room.component.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    meetingRooms.forEach(c => this.deleteMeetingRoom(c));
+  }
+
+  onFilterChange(value: string) {
+    if (!this.dataSource) {
+      return;
+    }
+    value = value.trim();
+    value = value.toLowerCase();
+    this.dataSource.filter = value;
```
Comment (aiandev) [inline client-side/src/app/_pages/meeting-room/meeting-room.component.ts]: @PavelGheorghe here what we discussed before , @hi-dima ignore this comment please

File: client-side/src/app/_pages/meeting-room/meeting-room.component.ts [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    this.isAllSelected()
+      ? this.selection.clear()
+      : this.dataSource.data.forEach(row => this.selection.select(row));
+  }
+
+  trackByProperty<T>(index: number, column: TableColumn<T>) {
+    return column.property;
+  }
+
+  ngOnDestroy() {}
```
Comment (aiandev) [inline client-side/src/app/_pages/meeting-room/meeting-room.component.ts]: remove unsed method , and make sure you unsubscribe all observable that you have in your component

File: client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts (hunk 2 above)
Comment (hi-dima) [reply client-side/src/app/_pages/meeting-room/meeting-room-create-update/meeting-room-create-update.component.ts]: I checked this line. We will have a bug if I delete this.

**Pull request merged**